### PR TITLE
Add parallel tests for rspec for dev use

### PR DESCRIPTION
**Pull request by asmega**
## Ticket and context

Ticket: none

- Adds ability to run rspec test suite in parallel
- This at the moment is for local developer use only as a first step
- Maybe after we can patch it into the CI process
- Has dropped down from 5/6 minutes runtime to about 90 seconds for my local machine

### Code quality checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests

### HTML Checks
- [x] All new pages have automated accessibility checks
- [x] All new pages have visual tests via Percy

### Gotchas
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?

- n/a

## External API changes

- none

### CODE REVIEW

File: config/environments/test.rb [hunk 1]
```diff
@@ -75,4 +75,8 @@
   # config.action_view.annotate_rendered_view_with_filenames = true
 
   config.cip_resource_bucket = "https://paas-s3-broker-prod-lon-ac28a7a5-2bc2-4d3b-8d16-a88eaef65526.s3.amazonaws.com"
+
+  if config.respond_to?(:web_console)
+    config.web_console.development_only = false
+  end
```
Comment (kicferk1) [inline config/environments/test.rb:81]: what's this for?

File: app/controllers/lead_providers/openapi_controller.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 # source: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/controllers/api_docs/vendor_api_docs/openapi_controller.rb
 
+require "lead_provider_api_specification"
+
```
Comment (kicferk1) [inline app/controllers/lead_providers/openapi_controller.rb:6]: Why do we need it here?

File: app/controllers/lead_providers/openapi_controller.rb [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 # source: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/controllers/api_docs/vendor_api_docs/openapi_controller.rb
 
+require "lead_provider_api_specification"
```
Comment (asmega) [inline app/controllers/lead_providers/openapi_controller.rb:5]: rails does not load `/lib` so should be required by hand is my understanding

**Review by kicferk1 — COMMENTED** [review]
Looks pretty good. I imagine it won't work in CI in the current form, right?

File: config/environments/test.rb [hunk 4]
```diff
@@ -75,4 +75,8 @@
   # config.action_view.annotate_rendered_view_with_filenames = true
 
   config.cip_resource_bucket = "https://paas-s3-broker-prod-lon-ac28a7a5-2bc2-4d3b-8d16-a88eaef65526.s3.amazonaws.com"
+
+  if config.respond_to?(:web_console)
+    config.web_console.development_only = false
```
Comment (asmega) [inline config/environments/test.rb:80]: seems to be some kind of quirk from `parallel_tests`. I don't actually know why but assume it for some reason it loads the dev gems whilst in test mode

Comment (asmega) [reply config/environments/test.rb:80]: actually maybe to do with spring and the way it loads

File: app/controllers/lead_providers/openapi_controller.rb (hunk 2 above)
Comment (asmega) [reply app/controllers/lead_providers/openapi_controller.rb:6]: prior to this `lead_provider_api_specification` is loaded by `app/controllers/lead_providers/guidance_controller.rb`. If that file is never loaded then `lead_provider_api_specification` will also not be available. `/lib` is not autoloaded by rails therefore must be required by hand before it is available

Comment (asmega) [reply app/controllers/lead_providers/openapi_controller.rb:6]: from `config/environments/test.rb` `config.eager_load = false` therefore not everything is loaded prior to test suite running.

Comment (kicferk1) [reply app/controllers/lead_providers/openapi_controller.rb:6]: Ah, alright, that makes sense

**Review by kicferk1 — APPROVED** [review]

**Comment by asmega** [comment]
> Looks pretty good. I imagine it won't work in CI in the current form, right?

It's not hooked up so won't work. I think we want to run it locally for now to iron out any weirdness before updating CI with potential flakiness.

**Pull request merged**